### DISCUSSION TITLE
项目部署到nginx后，post的接口全部405

### DISCUSSION THREAD

**Discussion by ideaviewes** [umijs/umi#10028] [Q&A] [answered]
项目部署到nginx后，post的接口全部405,之前完全没遇到这种问题。我看405的原因是因为用Post的方式请求了静态文件
我要怎么告诉nginx那个是接口，并不是静态文件。
我看网上的解决方案是把post转成get，这个对我这个场景不适用，有些接口必须用post。
nginx的配置文件如下
`server
{
    listen 80;
    listen 443 ssl http2;
    server_name admin.xxx.com;
    index index.php index.html index.htm default.php default.htm default.html;
    root /home/wwwroot/admin.xxx.com/dist;

    #SSL-START SSL相关配置，请勿删除或修改下一行带注释的404规则
    #error_page 404/404.html;
    ssl_certificate    /www/server/panel/vhost/cert/admin.xxx.com/fullchain.pem;
    ssl_certificate_key    /www/server/panel/vhost/cert/admin.xxx.com/privkey.pem;
    ssl_protocols TLSv1.1 TLSv1.2 TLSv1.3;
    ssl_ciphers EECDH+CHACHA20:EECDH+CHACHA20-draft:EECDH+AES128:RSA+AES128:EECDH+AES256:RSA+AES256:EECDH+3DES:RSA+3DES:!MD5;
    ssl_prefer_server_ciphers on;
    ssl_session_cache shared:SSL:10m;
    ssl_session_timeout 10m;
    add_header Strict-Transport-Security "max-age=31536000";
    error_page 497  https://$host$request_uri;
		#SSL-END

    #ERROR-PAGE-START  错误页配置，可以注释、删除或修改
    #error_page 404 /404.html;
    #error_page 502 /502.html;
    #ERROR-PAGE-END
    location / {
        root /home/wwwroot/admin.xxx.com/dist;
        try_files $uri /index.html;
    }
    location /api/ {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $host;
        proxy_pass https://www.xxx.com/admin/;
    }
}`
用的是umi 4

**Comment by fz6m** [comment, marked as answer]
把 `/api/` 的规则放到 `/` 的规则上面，`/` 被先命中了。

**Reply by ideaviewes** [reply]
嗯嗯，是这个问题，我这样解决了
![image](https://user-images.githubusercontent.com/5431843/206981330-1092b49b-a712-437c-a20b-229c2075f591.png)
